platform/base.js

```diff
@@ -15,44 +15,51 @@ class SayPlatformBase {
    * @param {number|null} speed Speed of text (e.g. 1.0 for normal, 0.5 half, 2.0 double)
    * @param {Function|null} callback A callback of type function(err) to return.
    */
-  speak (text, voice, speed, callback) {
-    if (typeof callback !== 'function') {
-      callback = () => {}
-    }
-
-    callback = once(callback)
+  async speak(text, voice, speed, callback) {
```

Comment:
Why have a callback if it's returns a promise?

Reply:
Backward compatible

Reply:
It hard to reason the order when callback() and say.then() get called. This might introduce a race condition.

Try adding a new async API that wraps the existing one.

Example:

```js
async speakAsync(text, voice, speed) {
    return new Promise((resolve, reject) => {
        speak(text, voice, speed, function callback(err) {
            if (err) {
                return reject(err);
            }

            resolve();
        })
    });
}
```